src/services/order_watcher_service.ts

```diff
@@ -116,6 +127,42 @@ export class OrderWatcherService {
                 }
                 break;
             }
+            case OrderWatcherLifeCycleEvents.PersistentUpdated: {
+                // Only update orders that exist in PersistentSignedOrders table. Avoid accidentally
+                // adding open orderbook orders to the table.
+
+                // 1. Filter out ignored
+                const filtered = orders.filter(
+                    apiOrder => !orderUtils.isIgnoredOrder(MESH_IGNORED_ADDRESSES, apiOrder),
+                );
+
+                // 2. Create the Update queries
+                // tslint:disable-next-line:promise-function-async
+                const updatePromises = filtered.map(apiOrder => {
+                    const { remainingFillableTakerAssetAmount, state: orderState, orderHash } = apiOrder.metaData;
+                    return this._connection.manager.update(PersistentSignedOrderEntity, orderHash, {
+                        remainingFillableTakerAssetAmount: remainingFillableTakerAssetAmount.toString(),
+                        orderState,
+                    });
+                });
+
+                // 3. Wait for results
+                await Promise.allSettled(updatePromises).then(results => {
+                    // Group by success or failure. Open orderbook orders should fail.
+                    const { fulfilled, rejected } = results.reduce(
+                        (acc, r, i) => {
+                            r.status === 'fulfilled' ? acc.fulfilled.push(i) : acc.rejected.push(i);
```

Comment:
NIT(feel free to ignore): maybe a bit more readable to store the object rather than the index the object has in the array?

Could even be a neat one-liner 😄 
```typescript
const [fulfilled, rejected] = _.partition(results, result => r.status === 'fulfilled');
```